### ISSUE TITLE
Self-contained AI

### ISSUE THREAD

**Issue by snakehill** [AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin#438]
Having just fought a hard and long battle to finally have a normally running ComfyUI be recognized by the plugin, I can't help but notice how much more tedious the installation and run process is compared to [krita-ai-diffusion](https://github.com/Acly/krita-ai-diffusion).

I understand wanting to use a global version of the AI, but having at least an option to have ComfyUI (or A1111) fully self-contained within Auto-Photoshop-StableDiffusion-Plugin would make the setup a lot easier, also for those who don't have full system control or wish to use it for only this one purpose.

Thank you for taking it into consideration!

**Comment by AbdullahAlfaraj** [comment]
what difficulties did you run into?

**Comment by snakehill** [comment]
> what difficulties did you run into?

Mostly that it couldn't find a running ComfyUI instance on the given IP on which it was running just fine, but it also threw some other errors along the way.

The Krita AI Diffusion plugin has most of the installation take place from the plugin itself, which creates a ComfyUI instance with models all within the plugin folder itself, which makes the installation much easier and more user-friendly. There also is no need to manually run a CPU or GPU instance in the background as that is done from within the plugin also.

The Auto-Photoshop-StableDiffusion-Plugin process is fine for those who already have A1111 or Comfy run on their system, especially when that is recognized by the plugin, but I wouldn't say it's for the average artist. Also on more locked-down systems, installing git (and the ComfyUI Manager) or running a global AI instance might not be possible, whereas doing all that through the plugin and its folder would solve that too (which should be doable without too much hassle in this case as shown by the Krita plugin).

Just a suggestion and idea to get this running on many more systems, as this exact thing made me install and stick by Krita for a good while. Your plugin and work is fantastic when (and now that) it's working otherwise!

**Comment by AbdullahAlfaraj** [comment]
@snakehill,

You’re right, Krita does offer more flexibility. Unfortunately, Photoshop’s restrictions make it impossible to start other software or install them from a UXP plugin. It’s a bummer, but that’s how it is. 😕

**Comment by snakehill** [comment]
That's a pity. If you find any way to make it easier (e.g. integrating the AI, ComfyUI or the Manager so that it wouldn't need git, whether or not via a separate, optional process), please do so, of course. Feel free to close this issue regardless.

Thank you for looking into it anyways and keep up the great work!
